key-vault.tf

```diff
@@ -6,6 +6,7 @@ resource "azurerm_key_vault" "tfvars" {
   sku_name                   = "standard"
   soft_delete_retention_days = 7
   enable_rbac_authorization  = false
+  purge_protection_enabled   = true
```

Comment:
We should check the behaviour of Terraform with this - If a `terraform destroy` fails because of this, we should make it conditional. Not a blocker on this PR though 👍

Reply:
From the resource source:
>Once Purge Protection has been Enabled it's not possible to Disable it. Support for [disabling purge protection is being tracked in this Azure API issue](https://github.com/Azure/azure-rest-api-specs/issues/8075). Deleting the Key Vault with Purge Protection Enabled will schedule the Key Vault to be deleted (which will happen by Azure in the configured number of days, currently 90 days - which will be configurable in Terraform in the future).